src/cmd/build.rs

```diff
@@ -337,62 +334,13 @@ fn optimize_wasm(
     })
 }
 
-/// Optimizes the Wasm supplied as `wasm` using the `binaryen-rs` dependency.
-///
-/// The supplied `optimization_level` denotes the number of optimization passes,
-/// resulting in potentially a lot of time spent optimizing.
-///
-/// If successful, the optimized wasm is written to `dest_optimized`.
-#[cfg(feature = "binaryen-as-dependency")]
-fn do_optimization(
-    dest_wasm: &OsStr,
-    dest_optimized: &OsStr,
-    optimization_level: OptimizationPasses,
-) -> Result<()> {
-    let mut dest_wasm_file = File::open(dest_wasm)?;
-    let mut dest_wasm_file_content = Vec::new();
-    dest_wasm_file.read_to_end(&mut dest_wasm_file_content)?;
-
-    let codegen_config = binaryen::CodegenConfig {
-        // Number of optimization passes (spends potentially a lot of time optimizing)
-        optimization_level: optimization_level.to_passes(),
-        // The default
-        shrink_level: optimization_level.to_shrink(),
-        // The default
-        debug_info: false,
-    };
-    log::info!(
-        "Optimization level passed to `binaryen` dependency: {}",
-        codegen_config.optimization_level
-    );
-    log::info!(
-        "Shrink level passed to `binaryen` dependency: {}",
-        codegen_config.shrink_level
-    );
-    let mut module = binaryen::Module::read(&dest_wasm_file_content)
-        .map_err(|_| anyhow::anyhow!("binaryen failed to read file content"))?;
-
-    if optimization_level != OptimizationPasses::Zero {
-        // binaryen-rs still uses the default optimization passes, even if zero
-        // is passed. this is the ticket for it: https://github.com/pepyakin/binaryen-rs/issues/56.
-        // we can remove the if condition here once the issue is fixed.
-        module.optimize(&codegen_config);
-    }
-
-    let mut optimized_wasm_file = File::create(dest_optimized)?;
-    optimized_wasm_file.write_all(&module.write())?;
-
-    Ok(())
-}
-
 /// Optimizes the Wasm supplied as `crate_metadata.dest_wasm` using
 /// the `wasm-opt` binary.
 ///
 /// The supplied `optimization_level` denotes the number of optimization passes,
 /// resulting in potentially a lot of time spent optimizing.
 ///
 /// If successful, the optimized wasm is written to `dest_optimized`.
-#[cfg(not(feature = "binaryen-as-dependency"))]
 fn do_optimization(
```

Comment:
Might be worth moving `do_optimization` back into `optimize_wasm` now that there are not 2 versions of it.

Reply:
I would prefer keeping it compartmentalized since `do_optimization` is already quite long and IMHO the code is more readable this way.